fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidResponseParser.java

```diff
@@ -38,20 +39,28 @@
      *
      * @param jsonResult  Druid results in json
      * @param schema  Schema for results
-     * @param queryType  the type of query
+     * @param queryType  the type of query, note that this implementation only supports instances of
+     * {@link DefaultQueryType}
      *
      * @return the set of results
      */
     public ResultSet parse(JsonNode jsonResult, ZonedSchema schema, QueryType queryType) {
 
         LOG.trace("Parsing druid query {} by json result: {} using schema: {}", queryType, jsonResult, schema);
 
+
+        if (!(queryType instanceof DefaultQueryType)) {
+            // Throw an exception for unsupported query types
+            unsupportedQueryType(queryType);
```

Comment:
We should add a note (likely to the changelog) that this test extension point doesn't support new types of queries.